aries_cloudagent/protocols/coordinate_mediation/v1_0/handlers/tests/test_keylist_handler.py

```diff
@@ -44,16 +46,20 @@ async def test_handler_no_active_connection(self, context):
             await handler.handle(context, responder)
             assert "inactive connection" in exc.value
 
-    async def test_handler_no_record(self, context, caplog):
-        caplog.set_level(logging.INFO)
-        handler, responder = KeylistHandler(), MockResponder()
-        await handler.handle(context, responder)
-        assert "not acting as mediator" in caplog.text
-        assert "Keylist received: " not in caplog.text
+    async def test_handler_no_record(self, context):
+        with async_mock.patch.object(
+            test_module, "LOG", async_mock.MagicMock()
+        ) as mock_logger:
+            handler, responder = KeylistHandler(), MockResponder()
+            await handler.handle(context, responder)
+            assert mock_logger.warning.call_count == 1
+            assert mock_logger.info.call_count == 0
```

Comment:
Can you elaborate on why these changes are necessary?

Reply:
If the caplog fixture isn't behaving as expected, I think it would be better to define our own `caplog` fixture in `conftest.py` instead of touching all these tests.